orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -147,13 +147,20 @@ def forward(ctx, *inputs, **kwargs):
                     # Assert that the input and model device match
                     _check_same_device(self._device, "Input argument to forward", *inputs)
 
+                    # training_io_binding and run_options are per InferenceSession call
+                    # TODO: we should try to reuse the output buffers as some of the output tensors are same sizes, expecially the backward graph outputs.
+                    training_io_binding = self._training_session.io_binding()
+                    run_options = C.RunOptions()
+                    
                     # Use IO binding
-                    _create_iobinding(self._training_io_binding, inputs, self._onnx_training, self._device)
+                    _create_iobinding(training_io_binding, inputs, self._onnx_training, self._device)
 
                     # Run and return module outputs.
-                    forward_outputs, run_id = self._training_session.run_forward(self._training_io_binding, self._run_options)
+                    forward_outputs, run_id = self._training_session.run_forward(training_io_binding, run_options)
                     user_outputs = tuple(_ort_output_to_torch_tensor(forward_output) for forward_output in forward_outputs)
                     ctx.run_id = run_id
+                    ctx.training_io_binding = training_io_binding
+                    ctx.run_options = run_options
```

Comment:
please add a comment here, 

adding run_options to ctx is not for used in the backward(), but more for keeping a reference count, so that it's not destroyed  after forward()

Reply:
@SherlockNoMad added comment
@thiagocrepaldi We use self.training_io_binding for io binding and self.run_options for run_options. Is that what you're talking about?